### DISCUSSION TITLE
Test results of [A2 VGA Card (RevB 1.4.0)] vs [A2 Heaven VGA card]...

### DISCUSSION THREAD

**Discussion by AncientBits** [markadev/AppleII-VGA#53] [General]
Hello everyone, new to this community!

Just bought this really nice, little A2 VGA card (Pico) for my Platinum IIe, which has been running A2 Heaven's VGA Card piggy-backed on RAM-Works IIII from ReactiveMicro.

I absolutely **LOVE** the crisp, per-pixel image rendition of the A2-VGA Pico on my Viewsonic VP-930b, as well as the excellent  support of Video-7's display modes AND Apple Desktop compatibility... MUCH better than my A2-Heaven, no doubt!!

But here's where I've found some (deeper) problems: when launching French Touch's Crazy Cycles (65C02) and Scroll, Scroll, Scroll demos, the A2-Pico almost fully FAILS executing these demos (I have simultaneous composite-out via video-processor, and I can see that on-board legacy video does displays everything correctly). My A2-Heaven also displays everything correctly. (NOTE: I did fully disabled FastChip accelerator when testing both...)

What would be the problem here?

Thanks everyone!!

**Comment by markadev** [comment]
Hey there!

I'm not familiar with those programs but from the names it sounds like you're hitting a limitation with the current firmware.  Basically, the VGA card's video output is not synchronized with the Apple II's built-in video memory scanner so programs that explicitly synchronize with the video memory scanner and cycle-count to switch video modes in the middle of a frame will not really work.

Your other VGA card that connected via the Aux slot gets the video data directly from the video memory scanner as it scans so it probably synchronizes naturally.

Some more recent hardware versions of my VGA card have access to the video SYNC signal so theoretically this could be supported but I'm not really planning on implementing that.

**Reply by AncientBits** [reply]
Thanks for your reply!

Well, that explains my suspicions... The Apple II had a pretty good French scene (in multiple domains), and some of the work from "French Touch" is a real "tour the force" on the A2/A2e. A good sampler of their here: http://fr3nch.t0uch.free.fr/ , and some visuals: https://www.youtube.com/watch?v=HtGhfb4_yAE

ALSO, I noticed that on a much more known title (A2osX), the splash screen gets (horizontally) corrupted at the end of Stage-3 boot (or so), right before entering the Console-session (final)... Have you seen it? May be also worth looking at it..

(Last but not least: I posted on the Ideas forum about embedding SmartPort rom-support on the A2 VGA (the code is already available in SoftSP and DiY roms for Apple Serial or Gappler cards). Is it possible?)

Cheers!

**Reply by markadev** [reply]
Ah that's cool.  I never saw any Apple II demos but I thought that they must exist somewhere 😃.

I took a look at A2osx and didn't see any screen corruption in the boot sequence like you mentioned.  Specifically stage 3 looks OK, then the screen goes blank, and then the GeTTY output starts.  This is on my Platinum IIe with no extra hardware beyond the disk & VGA cards.  I'm not too familiar with accelerators so I suppose that if you have an accelerator installed then it might introduce a small timing variation, even when disabled, that causes the VGA card to mis-read the bus data.  A2osx is pretty wild - I never saw it before and it looks impressive so thanks for sharing.

(I'll respond to your other thread too)

**Reply by AncientBits** [reply]
Well, It seems that I've found a plausible reason for my A2osX splash-screen corruption: Uthernet2 in Slot-1, FastChip in Slot-2 and A2VGA Pico in Slot-3... FastChip has its on-board RAMworks enabled, and defaults to slot-3, which is caught and setup there by A2osX's RAMworks driver during Stage3, or so... I will triple-check by moving A2VGA somewhere else momentarily and verify that corruption disappears... 

Cheers!
